yarn-project/prover-node/src/monitors/epoch-monitor.ts

```diff
@@ -22,6 +37,11 @@ export class EpochMonitor implements Traceable {
 
   private latestEpochNumber: bigint | undefined;
 
+  private epochs: EpochStatus[] = [];
+  private blockHashes = new Map<number, string>();
+  private l2Tips: L2Tips | undefined = undefined;
```

Comment:
The `L2TipsStore` encapsulates most of the work for keeping the "local" view of the data in sync. Maybe using it here would remove some code from this class.